Sources/SwiftLexicalLookup/Scopes/ScopeImplementations.swift

```diff
@@ -503,17 +415,46 @@ import SwiftSyntax
           checkIdentifier(identifier, refersTo: name, at: lookUpPosition)
         }
 
-      return (implicitSelf.isEmpty ? [] : [.fromScope(self, withNames: implicitSelf)]) + [.lookInMembers(self)]
-        + defaultLookupImplementation(identifier, at: lookUpPosition, with: config)
+      return LookupResult.getResultArray(for: self, withNames: implicitSelf)
+        + [.lookInGenericParametersOfExtendedType(self)]
+        + defaultLookupImplementation(identifier, at: lookUpPosition, with: config, propagateToParent: false)
+        + [.lookInMembers(self)]
+        + lookupInParent(identifier, at: lookUpPosition, with: config)
+    } else if !extendedType.range.contains(lookUpPosition) && genericWhereClause != nil {
```

Comment:
Okay, it's a stylistic thing, but I prefer that an "if" ending with a "return" not have an "else" following it, because it means unnecessary nesting. So instead of

```swift
if x {
 // ...
 return y
} else {
  // rest of body
}
```

we de-nest as:

```swift
if x {
  // ...
  return y
}

// rest of body
```

Reply:
That's a good idea. I de-nested those couple of `else` statements inside `ExtensionDeclSyntax` scope extension.